src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -185,6 +185,9 @@ The .NET Foundation licenses this file to you under the MIT license.
       <_IlcConditionallyRootedAssemblies Include="@(ManagedAssemblyToLink)" Condition="%(ManagedAssemblyToLink.TrimMode) == 'copyused'" />
       <_IlcTrimmedAssemblies Include="@(ManagedAssemblyToLink)" Condition="%(ManagedAssemblyToLink.TrimMode) == 'link'" />
       <_IlcNoSingleWarnAssemblies Include="@(ManagedAssemblyToLink)" Condition="%(ManagedAssemblyToLink.TrimmerSingleWarn) == 'false'" />
+
+      <!-- always export unmanaged entry points from System.Private.CoreLib -->
+      <UnmanagedEntryPointsAssembly Include="System.Private.CoreLib" />
```

Comment:
Could you move this above around the ItemGroup that sets `<AutoInitializedAssemblies Include="System.Private.CoreLib" />`? It's better not to modify the ItemGroup within a Target if we can help it, plus we want this to be conditioned on `$(IlcSystemModule) == ''` so that we don't break scenarios that do things like https://github.com/MichalStrehovsky/zerosharp.

Reply:
Done.